modules/dmrpp_module/ngap_container/NgapOwnedContainer.cc

```diff
@@ -559,6 +568,49 @@ bool NgapOwnedContainer::get_dmrpp_from_cache_or_remote_source(string &dmrpp_str
     return true;
 }
 
+static inline bool is_eq(const char *value, const char *key) {
+    return strcmp(value, key) == 0;
+}
+
+// TODO-docstring
+// Implementation borrowed from DMZ::process_dataset
+NgapApi::DataAccessUrls NgapOwnedContainer::extract_s3_data_urls_from_dmrpp(const string &dmrpp_string) {
+    // If the dmrpp is invalid, we will have hit a failure before now---so we can assume
+    // it's generally safe---so do basically no additional safety checking
+
+    // Load the xml document
+    pugi::xml_document result_xml_doc;
+    pugi::xml_parse_result result = result_xml_doc.load_string(dmrpp_string.c_str(), pugi::parse_default | pugi::parse_ws_pcdata_single);
```

Comment:
The variable `result` is never checked. Does that matter? Do we care??

Reply:
Weird, I was *sure* that I already responded to this comment somewhere. Apologies for the duplication if I'm just missing it!

Anyway, yes: I'll add a check here, that's a smart thing to do! Also, it would be fairly unlikely to fail at this point, as this is the second time we've parsed this same dmrpp_string, and upon failure the earlier time we'd be blocked from ever reaching this point. 

- [ ] Check result here